pennylane/labs/tests/resource_estimation/ops/op_math/test_symbolic.py

```diff
@@ -0,0 +1,230 @@
+
+        expected = re.Resources(gate_types={tracking_name: 1})
+        gate_set = {tracking_name}
+
+        assert re.get_resources(op, gate_set=gate_set) == expected
+
+
+class TestResourceControlled:
+    """Tests for ResourceControlled"""
+
+    controlled_ops = [
+        re.ResourceControlled(re.ResourceQFT([0, 1]), control_wires=[2]),
+        re.ResourceControlled(
+            re.ResourceControlled(re.ResourceQFT([0, 1]), control_wires=[2]), control_wires=[3]
+        ),
+        re.ResourceControlled(re.ResourceQFT([0, 1]), control_wires=[2, 3], control_values=[0, 1]),
+        re.ResourceControlled(
+            re.ResourceAdjoint(re.ResourceQFT([0, 1])),
+            control_wires=[2, 3],
+            control_values=[0, 1],
+            work_wires=[4],
+        ),
+    ]
+
+    expected_params = [
+        {
+            "base_class": re.ResourceQFT,
+            "base_params": {"num_wires": 2},
+            "num_ctrl_wires": 1,
+            "num_ctrl_values": 0,
+            "num_work_wires": 0,
+        },
+        {
+            "base_class": re.ResourceControlled,
+            "base_params": {
+                "base_class": re.ResourceQFT,
+                "base_params": {"num_wires": 2},
+                "num_ctrl_wires": 1,
+                "num_ctrl_values": 0,
+                "num_work_wires": 0,
+            },
+            "num_ctrl_wires": 1,
+            "num_ctrl_values": 0,
+            "num_work_wires": 0,
+        },
+        {
+            "base_class": re.ResourceQFT,
+            "base_params": {"num_wires": 2},
+            "num_ctrl_wires": 2,
+            "num_ctrl_values": 1,
+            "num_work_wires": 0,
+        },
+        {
+            "base_class": re.ResourceAdjoint,
+            "base_params": {"base_class": re.ResourceQFT, "base_params": {"num_wires": 2}},
+            "num_ctrl_wires": 2,
+            "num_ctrl_values": 1,
+            "num_work_wires": 1,
+        },
+    ]
+
+    @pytest.mark.parametrize("op, expected", zip(controlled_ops, expected_params))
+    def test_resource_params(self, op, expected):
+        """Test that the resources are correct"""
+        assert op.resource_params() == expected
+
+    expected_names = [
+        "C(QFT,1,0,0)",
+        "C(C(QFT,1,0,0),1,0,0)",
+        "C(QFT,2,1,0)",
+        "C(Adjoint(QFT),2,1,1)",
+    ]
+
+    @pytest.mark.parametrize("op, expected", zip(controlled_ops, expected_names))
+    def test_tracking_name(self, op, expected):
+        """Test that the tracking name is correct"""
+        name = op.tracking_name_from_op()
+        assert name == expected
+
+    @pytest.mark.parametrize("op", controlled_ops)
+    def test_tracking(self, op):
+        """Test that adjoints can be tracked."""
+        tracking_name = op.tracking_name_from_op()
+
+        expected = re.Resources(gate_types={tracking_name: 1})
+        gate_set = {tracking_name}
+
+        assert re.get_resources(op, gate_set=gate_set) == expected
```

Comment:
Same comment as above, your expected resources object should have correct num_gates and wires